### PR TITLE
Link MKL to libknowhere

### PR DESCRIPTION
**Pull request by matchyc**
Signed-off-by: matchyc <dawnlight.yc@protonmail.com>

This link action will be triggered when building Knowhere with DiskANN, otherwise, it won't happen.

### CODE REVIEW

**Comment by matchyc** [comment]
not ready /unassign @cydrain

**Comment by matchyc** [comment]
not ready

**Comment by matchyc** [comment]
@Presburger

File: cmake/ThirdPartyPackagesKw.cmake [hunk 1]
```diff
@@ -20,6 +20,21 @@ if ( LINUX AND KNOWHERE_WITH_DISKANN)
...
         )
+
+    # For Ubuntu 18.04 (install MKL by script)
+    set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${MKL_DIR})
+    set(MKL_ROOT "/opt/intel/oneapi/mkl/latest")
+    set(OMP_PATH "/opt/intel/oneapi/compiler/2022.0.2/linux/compiler/lib/intel64_lin/")
+    find_package(MKL REQUIRED)
+
+    # For Ubuntu 20.04/22.04 (install MKL by apt)
+    link_directories(/usr/lib/x86_64-linux-gnu/mkl)
```
Comment (cydrain) [inline cmake/ThirdPartyPackagesKw.cmake:31]: can the path hard-coded work in different platforms ?

Comment (Presburger) [reply cmake/ThirdPartyPackagesKw.cmake:31]: ubuntu works, we will replace mkl to openblas in the future.

**Pull request merged**